compiler/rustc_query_system/src/query/mod.rs

```diff
@@ -25,15 +25,20 @@ use rustc_errors::Diagnostic;
 use rustc_span::def_id::DefId;
 use rustc_span::Span;
 
+/// Description of a frame in the query stack.
+///
+/// This is mostly used in case of cycles for error reporting.
 #[derive(Clone, Debug)]
-pub struct Query {
+pub struct QueryStackFrame {
     pub name: &'static str,
     pub description: String,
     span: Option<Span>,
+    /// This hash is used to deterministically pick
+    /// a query to remove cycles in the parallel compiler.
```

Comment:
Should this be cfg(parallel_compiler) then? Probably doesn't actually matter in practice since presumably it's not hit at all in successful compilations?